pages/front.js

```diff
@@ -1,31 +1,60 @@
+import { Component } from 'react'
 import { compose } from 'react-apollo'
+
+import { routes } from '../lib/routes'
 import Frame from '../components/Frame'
 import Front from '../components/Front'
+import StatusError from '../components/StatusError'
+
 import withData from '../lib/apollo/withData'
 import withMembership, { UnauthorizedPage } from '../components/Auth/withMembership'
-import StatusError from '../components/StatusError'
+import withInNativeApp from '../lib/withInNativeApp'
 
 const KNOWN_PATHS = ['/feuilleton']
 
-const FrontPage = (props) => {
-  if (props.isMember) {
-    return <Front {...props} />
-  }
-  const { url } = props
-  if (KNOWN_PATHS.indexOf(url.asPath.split('?')[0]) !== -1) {
-    return <UnauthorizedPage {...props} />
+const isPathKnown = (url) => {
+  return KNOWN_PATHS.indexOf(url.asPath.split('?')[0]) !== -1
+}
+
+class FrontPage extends Component {
+  render () {
+    const { url, isMember, inNativeIOSApp, serverContext } = this.props
+
+    if (isMember) {
+      return <Front {...this.props} />
+    }
+
+    if (isPathKnown(url)) {
+      if (serverContext && !inNativeIOSApp) {
```

Comment:
you should still handle the client case:
https://github.com/orbiting/republik-frontend/blob/3e13bd2d774b7d13e3c33fa49da6c0ba700d85ff/components/StatusError/index.js#L73
_client side links could still arrive without `serverContext`, while there shouldn't be any client side links for non members it may still happen and rendering a 404 for `/feuilleton` should always be prevented_

Reply:
And contrary to previous review: use `pushRoute` instead of `replaceRoute`. Because one could be not logged in, arrive at `/feuilleton`, be redirect to `/`, login and want to press back to get to `/feuilleton`.